From: Tony Jones <tonyj@suse.de>
Subject: Fix LDFLAGS order in utils/Makefile
References: none
Upstream: submitted Oct/2/2019
Signed-off-by: Tony Jones <tonyj@suse.de>

Binutils changed some time ago such that -l<dynamic lib> needs to follow the
same existing rules as -l<static lib>, namely to be specified after any .o/.a
files.

--- a/src/utils/Makefile	2017-12-19 18:14:32.000000000 +0100
+++ b/src/utils/Makefile	2018-08-22 15:56:27.704153834 +0200
@@ -16,46 +16,46 @@
 default all utils: $(ALL)
 
 papi_avail: papi_avail.o $(PAPILIB) print_header.o
-	$(CC) $(LDFLAGS) -o papi_avail papi_avail.o print_header.o $(PAPILIB)
+	$(CC) -o papi_avail papi_avail.o print_header.o $(PAPILIB) $(LDFLAGS)
 
 papi_clockres: papi_clockres.o $(PAPILIB) $(CLOCKCORE)
-	$(CC) $(LDFLAGS) -o papi_clockres papi_clockres.o $(PAPILIB) $(CLOCKCORE) -lm
+	$(CC) -o papi_clockres papi_clockres.o $(PAPILIB) $(CLOCKCORE) $(LDFLAGS) -lm
 
 papi_command_line: papi_command_line.o $(PAPILIB) $(DOLOOPS)
-	$(CC) $(LDFLAGS) -o papi_command_line papi_command_line.o $(PAPILIB) $(DOLOOPS)
+	$(CC) -o papi_command_line papi_command_line.o $(PAPILIB) $(DOLOOPS) $(LDFLAGS)
 
 papi_component_avail: papi_component_avail.o $(PAPILIB) print_header.o
-	$(CC) $(LDFLAGS) -o papi_component_avail papi_component_avail.o $(PAPILIB) print_header.o
+	$(CC) -o papi_component_avail papi_component_avail.o $(PAPILIB) print_header.o $(LDFLAGS)
 
 papi_cost: papi_cost.o $(PAPILIB) cost_utils.o
-	$(CC) $(LDFLAGS) -o papi_cost papi_cost.o cost_utils.o $(PAPILIB) -lm
+	$(CC) -o papi_cost papi_cost.o cost_utils.o $(PAPILIB) $(LDFLAGS) -lm
 
 papi_decode: papi_decode.o $(PAPILIB)
-	$(CC) $(LDFLAGS) -o papi_decode papi_decode.o $(PAPILIB)
+	$(CC) -o papi_decode papi_decode.o $(PAPILIB) $(LDFLAGS)
 
 papi_error_codes: papi_error_codes.o $(PAPILIB)
-	$(CC) $(LDFLAGS) -o papi_error_codes papi_error_codes.o $(PAPILIB)
+	$(CC) -o papi_error_codes papi_error_codes.o $(PAPILIB) $(LDFLAGS)
 
 papi_event_chooser: papi_event_chooser.o $(PAPILIB) print_header.o
-	$(CC) $(LDFLAGS) -o papi_event_chooser papi_event_chooser.o print_header.o $(PAPILIB)
+	$(CC) -o papi_event_chooser papi_event_chooser.o print_header.o $(PAPILIB) $(LDFLAGS)
 
 papi_hybrid_native_avail: papi_hybrid_native_avail.o $(PAPILIB)
-	$(CC) $(LDFLAGS) -o papi_hybrid_native_avail papi_hybrid_native_avail.o $(PAPILIB)
+	$(CC) -o papi_hybrid_native_avail papi_hybrid_native_avail.o $(PAPILIB) $(LDFLAGS)
 
 papi_mem_info: papi_mem_info.o $(PAPILIB)
-	$(CC) $(LDFLAGS) -o papi_mem_info papi_mem_info.o $(PAPILIB)
+	$(CC) -o papi_mem_info papi_mem_info.o $(PAPILIB) $(LDFLAGS)
 
 papi_multiplex_cost: papi_multiplex_cost.o $(PAPILIB) cost_utils.o
-	$(CC) $(LDFLAGS) -o papi_multiplex_cost papi_multiplex_cost.o cost_utils.o $(PAPILIB) -lm
+	$(CC) -o papi_multiplex_cost papi_multiplex_cost.o cost_utils.o $(PAPILIB) $(LDFLAGS) -lm
 
 papi_native_avail: papi_native_avail.o $(PAPILIB) print_header.o
-	$(CC) $(LDFLAGS) -o papi_native_avail papi_native_avail.o $(PAPILIB) print_header.o
+	$(CC) -o papi_native_avail papi_native_avail.o $(PAPILIB) print_header.o $(LDFLAGS)
 
 papi_version: papi_version.o $(PAPILIB)
-	$(CC) $(LDFLAGS) -o papi_version papi_version.o $(PAPILIB)
+	$(CC) -o papi_version papi_version.o $(PAPILIB) $(LDFLAGS)
 
 papi_xml_event_info: papi_xml_event_info.o $(PAPILIB)
-	$(CC) $(LDFLAGS) -o papi_xml_event_info papi_xml_event_info.o $(PAPILIB)
+	$(CC) -o papi_xml_event_info papi_xml_event_info.o $(PAPILIB) $(LDFLAGS)
 
 
 
